src/features/lightspeed/inlineSuggestions.ts

```diff
@@ -841,5 +840,61 @@ export async function inlineSuggestionUserActionHandler(
     inlineSuggestionFeedbackPayload,
     lightSpeedManager.orgTelemetryOptOut
   );
+  resetSuggestionData();
+}
+
+function inlineSuggestionPending(): boolean {
+  if (vscode.window.activeTextEditor?.document.languageId !== "ansible") {
+    return false;
+  }
+  const editor = vscode.window.activeTextEditor;
+  if (!editor) {
+    return false;
+  }
+  if (!inlineSuggestionData["suggestionId"]) {
+    return false;
+  }
+  return true;
+}
+
+function resetSuggestionData(): void {
   inlineSuggestionData = {};
+  insertTexts = [];
+}
+
+export async function inlineSuggestionTextDocumentChangeHandler(
+  e: vscode.TextDocumentChangeEvent
+) {
+  // Exit early if Lightspeed or Inline suggestion is not enabled.
+  const lightSpeedSetting =
+    lightSpeedManager.settingsManager.settings.lightSpeedService;
+  if (!lightSpeedSetting.enabled || !lightSpeedSetting.suggestions.enabled) {
+    return;
+  }
+
+  // If the user accepted a suggestion on the widget, ansible.lightspeed.inlineSuggest.accept
+  // command is not sent. This method checks if a text change that matches to the current
+  // suggestion was found. If such a change was detected, we assume that the user accepted
+  // the suggestion on the widget.
+  if (
+    inlineSuggestionPending() &&
+    insertTexts &&
+    e.document.languageId === "ansible" &&
+    e.contentChanges.length > 0
+  ) {
+    const suggestionId = inlineSuggestionData["suggestionId"];
+
+    e.contentChanges.forEach(async (c) => {
+      if (c.text === insertTexts[0]) {
```

Comment:
Also can we reuse the match logic to check is suggestion is coming from lighspeed and not from any other provider by adding the check here https://github.com/ansible/vscode-ansible/blob/main/src/features/lightspeed/inlineSuggestions.ts#L772. This check will also solve the issue of contentmatches API triggering for suggestion not coming from lightspeed.

If the document change handler is not triggered after hitting TAB within the `inlineSuggestionCommitHandler` method another approach is to compare the document content before the inline suggestion trigger request with the document content after inline suggestion is accepted and the diff between the two should be same as the one received in lightspeed completion response.

Reply:
At https://github.com/ansible/vscode-ansible/blob/main/src/features/lightspeed/inlineSuggestions.ts#L772 , is there a handy way to see which text is being inserted with the commit command? If it is possible to get the currently displayed text, I think it is a good idea to add the match logic there...

Reply:
I was not able to get the inserted text in the commit command handler through a callback approach but was to get the entire file content with inserted text using below snippet in commit command handler
```
const activeEditor = vscode.window.activeTextEditor;

if (activeEditor) {
    const document = activeEditor.document;
    const text = document.getText();
    console.log(text);
} 
```

That's the reason I mentioned another approach is to compare the document content before the inline suggestion trigger request with the document content after inline suggestion is accepted and the diff between the two should be same as the one received in lightspeed completion response in the earlier comment.

I think this comment can be handled in the follow-up PR

Reply:
Ganesh, thanks for the explanation. Ok, let's handle this in the follow-up PR.